--- a/configure.ac	2026-03-04 01:20:57.008128369 +0100
+++ b/configure.ac	2026-03-04 01:21:18.264845126 +0100
@@ -11961,7 +11961,7 @@
 if test "$with_system_zxcvbn" = "yes"; then
     AC_MSG_RESULT([external])
     SYSTEM_ZXCVBN=TRUE
-    AC_CHECK_HEADER(zxcvbn.h, [],
+    AC_CHECK_HEADER(zxcvbn/zxcvbn.h, [],
        [ AC_MSG_ERROR(zxcvbn headers not found.)], [])
     AC_CHECK_LIB(zxcvbn, ZxcvbnMatch, [],
         [ AC_MSG_ERROR(zxcvbn library not found.)], [])
--- a/svl/source/misc/PasswordHelper.cxx	2026-03-04 01:21:27.538667335 +0100
+++ b/svl/source/misc/PasswordHelper.cxx	2026-03-04 01:21:42.787360137 +0100
@@ -25,7 +25,7 @@
 #include <unicode/regex.h>
 #include <unicode/unistr.h>
 #include <unicode/errorcode.h>
-#include <zxcvbn.h>
+#include <zxcvbn/zxcvbn.h>
 #include <sal/log.hxx>
 
 using namespace com::sun::star;
