--- a/cmake/cef_variables.cmake	2024-02-22 19:38:49.641655448 +0100
+++ b/cmake/cef_variables.cmake	2024-02-22 19:39:11.207811568 +0100
@@ -125,7 +125,7 @@
     -pthread                        # Use the pthread library
     -Wl,--disable-new-dtags         # Don't generate new-style dynamic tags in ELF
     -Wl,--fatal-warnings            # Treat warnings as errors
-    -Wl,-rpath,.                    # Set rpath so that libraries can be placed next to the executable
+    #-Wl,-rpath,.                    # Set rpath so that libraries can be placed next to the executable
     -Wl,-z,noexecstack              # Mark the stack as non-executable (security feature)
     -Wl,-z,now                      # Resolve symbols on program start instead of on first use (security feature)
     -Wl,-z,relro                    # Mark relocation sections as read-only (security feature)
