--- a/configure.ac	2024-03-04 13:10:16.002107721 +0100
+++ b/configure.ac	2024-03-04 13:13:00.789204353 +0100
@@ -614,6 +614,24 @@
 	]
 )
 
+AC_MSG_CHECKING([whether to enable StormByte branding])
+AC_ARG_ENABLE(stormbyte,
+	[  --enable-stormbyte       enable StormByte branding],[
+	if test "$enableval" = "yes"; then
+		AC_DEFINE_UNQUOTED([OS_URL],"http://blog.stormbyte.org/")
+		AC_DEFINE_UNQUOTED([ROOTDEV_MANUFACTURERURL],"http://blog.stormbyte.org/")
+		AC_DEFINE_UNQUOTED([ROOTDEV_MANUFACTURER],"SormByte")
+		AC_DEFINE_UNQUOTED([ROOTDEV_MODELNAME],"StormByte Cube")
+		AC_DEFINE_UNQUOTED([ROOTDEV_MODELDESCRIPTION],"Cube")
+		AC_MSG_RESULT([yes])
+	else
+		AC_MSG_RESULT([no])
+	fi
+	],[
+		AC_MSG_RESULT([no])
+	]
+)
+
 AC_MSG_CHECKING([whether to build a static binary executable])
 AC_ARG_ENABLE(static,
 	[  --enable-static         build a static binary executable],[
