--- a/gtk/configure.ac	2020-06-09 11:43:56.320064837 +0200
+++ b/gtk/configure.ac	2020-06-09 11:44:09.206065356 +0200
@@ -199,7 +199,7 @@
 
 AM_CONDITIONAL([MINGW], [test "x$mingw_flag" = "xyes"])
 
-HB_LIBS="$HB_LIBS -lhandbrake -lavformat -lavfilter -lavcodec -lavutil -ldav1d -lswresample -lpostproc -ldvdnav -ldvdread -lmp3lame -lvorbis -lvorbisenc -logg -lswscale -ltheoraenc -ltheoradec -lvpx -lz -lbz2 -lbluray -lass -lfontconfig -lfreetype -lxml2 -ljansson -lopus -lspeex -llzma"
+HB_LIBS="$HB_LIBS -lhandbrake -lavformat -lavfilter -lavcodec -lavutil -ldav1d -lswresample -lpostproc -ldvdnav -ldvdread -lmp3lame -lvorbis -lvorbisenc -logg -lx264 -lswscale -ltheoraenc -ltheoradec -lvpx -lz -lbz2 -lbluray -lass -lfontconfig -lfreetype -lxml2 -ljansson -lopus -lspeex -llzma -ldl"
 HB_CPPFLAGS="$HB_CPPFLAGS $HBINC"
 
 PKG_CHECK_MODULES([x264], [x264], sys_x264=yes, sys_x264=no)
--- a/test/module.defs	2020-06-09 11:45:18.568068147 +0200
+++ b/test/module.defs	2020-06-09 11:45:49.014069373 +0200
@@ -69,6 +69,9 @@
 ifeq (1, $(FEATURE.numa))
     TEST.GCC.l += numa
 endif
+ifeq (1, $(FEATURE.x265))
+    TEST.GCC.l += x265
+endif
 else ifeq ($(HOST.system),kfreebsd)
     TEST.GCC.l += pthread dl m
 else ifeq ($(HOST.system),freebsd)
