--- a/llvm/CMakeLists.txt.orig	2020-07-21 22:07:22.563051108 +0200
+++ b/llvm/CMakeLists.txt	2020-07-21 22:07:42.957051298 +0200
@@ -322,6 +322,7 @@
 endif()
 
 option(LLVM_ENABLE_UNWIND_TABLES "Emit unwind tables for the libraries" ON)
+set(UNWIND_LIBRARIES "" CACHE STRING "Additional libraries needed for unwind functions")
 
 option(LLVM_ENABLE_CRASH_OVERRIDES "Enable crash overrides." ON)
 if(LLVM_ENABLE_CRASH_OVERRIDES)
--- a/llvm/cmake/modules/AddLLVM.cmake.orig	2020-07-21 22:08:01.385051471 +0200
+++ b/llvm/cmake/modules/AddLLVM.cmake	2020-07-21 22:12:54.741054211 +0200
@@ -610,6 +610,9 @@
       ${lib_deps}
       ${llvm_libs}
       )
+  if(UNWIND_LIBRARIES)
+    target_link_libraries(${name} PRIVATE -Wl,--push-state -Wl,--as-needed ${UNWIND_LIBRARIES} -Wl,--pop-state)
+  endif()
 
   if(LLVM_COMMON_DEPENDS)
     add_dependencies(${name} ${LLVM_COMMON_DEPENDS})
@@ -837,6 +840,9 @@
     # API for all shared libaries loaded by this executable.
     target_link_libraries(${name} PRIVATE ${LLVM_PTHREAD_LIB})
   endif()
+  if(UNWIND_LIBRARIES)
+    target_link_libraries(${name} PRIVATE -Wl,--push-state -Wl,--as-needed ${UNWIND_LIBRARIES} -Wl,--pop-state)
+  endif()
 
   llvm_codesign(${name} ENTITLEMENTS ${ARG_ENTITLEMENTS} BUNDLE_PATH ${ARG_BUNDLE_PATH})
 endmacro(add_llvm_executable name)
--- a/llvm/unittests/Support/DynamicLibrary/CMakeLists.txt.orig	2020-07-21 22:13:39.504054629 +0200
+++ b/llvm/unittests/Support/DynamicLibrary/CMakeLists.txt	2020-07-21 22:14:35.400055152 +0200
@@ -26,6 +26,9 @@
     PipSqueak.cpp
     )
   set_target_properties(${NAME} PROPERTIES FOLDER "Tests")
+  if(UNWIND_LIBRARIES)
+    target_link_libraries(${NAME} PRIVATE -Wl,--push-state -Wl,--as-needed ${UNWIND_LIBRARIES} -Wl,--pop-state)
+  endif()
 
   set_output_directory(${NAME}
     BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}
